bots/twitterbot.go

```diff
@@ -0,0 +1,155 @@
+package bots
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/coniks-sys/coniks-go/keyserver"
+	p "github.com/coniks-sys/coniks-go/protocol"
+	"github.com/dghubble/go-twitter/twitter"
+	"github.com/dghubble/oauth1"
+)
+
+type TwitterBot struct {
+	client        *twitter.Client
+	stream        *twitter.Stream
+	coniksAddress string
+	handle        string
+}
+
+var _ Bot = (*TwitterBot)(nil)
+
+type TwitterConfig struct {
+	CONIKSAddress string `toml:"coniks_address"`
+	TwitterOAuth  `toml:"twitter_oauth"`
+	Handle        string `toml:"twitter_bot_handle"`
+}
+
+type TwitterOAuth struct {
+	ConsumerKey    string
+	ConsumerSecret string
+	AccessToken    string
+	AccessSecret   string
+}
+
+func NewTwitterBot(path string) (Bot, error) {
+	var conf TwitterConfig
+	if _, err := toml.DecodeFile(path, &conf); err != nil {
+		return nil, fmt.Errorf("Failed to load config: %v", err)
+	}
+
+	// Notify if the CONIKS key server is down
+	if !testCONIKSConnection(conf.CONIKSAddress) {
```

Comment:
Ya, stat seems less intrusive.  Let's do that.

Ultimately, we may want to reconsider what I said in #75.  Maybe opening a long-lived connection at the start with the keyserver and doing some "first byte sent indicates how many more bytes to read" protocol is the way to go.

Opening a issue to discuss that may be good.

We could use the same protocol for the public connections but close them after responding, rather than a loop.

Anyways, to discuss ...